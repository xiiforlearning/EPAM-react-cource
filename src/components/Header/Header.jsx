import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Button } from '../../common/Button/Button';
import { BUTTON_TEXT_LOGOUT } from '../../constants';
import { Logo } from './components/Logo/Logo';
import styles from './header.module.scss';

const Header = ({ name }) => {
	const navigate = useNavigate();

	return (
		<header className={styles.headerWrapper}>
			<Logo></Logo>
			{!!name && (
				<div className={styles.headerItem}>
					<p>{name}</p>
					<Button
						text={BUTTON_TEXT_LOGOUT}
						onClick={() => {
							localStorage.removeItem('token');
							navigate('/login');
							window.dispatchEvent(new Event('token'));
						}}
					></Button>
				</div>
			)}
		</header>
	);
};

export default Header;
