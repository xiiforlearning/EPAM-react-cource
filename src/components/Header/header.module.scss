.headerWrapper{
    display: flex;
    justify-content: space-between;
    width: 100%;
    margin: 0 auto;
    padding: 10px 20px;
    align-items: center;
    border: 1px solid red;
}
.headerItem {
    display: flex;
    flex-direction: row;
    gap: 30px;
}