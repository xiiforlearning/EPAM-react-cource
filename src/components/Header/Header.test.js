import React from 'react';
import { create } from 'react-test-renderer';
import Header from './Header';

const mockedUsedNavigate = jest.fn();
const mockedUsedSelector = jest.fn();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockedUsedNavigate,
}));

jest.mock('react-redux', () => ({
	...jest.requireActual('react-redux'),
	useSelector: () => mockedUsedSelector,
}));

describe('Header test', () => {
	it('should render Header component properly', () => {
		const component = create(<Header />);

		expect(component.toJSON()).toMatchSnapshot();
	});
});
