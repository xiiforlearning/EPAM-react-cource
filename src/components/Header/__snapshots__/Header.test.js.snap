// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test should render Header component properly 1`] = `
<header
  className="headerWrapper"
>
  <img
    alt="Logo"
    src="logo.png"
  />
  <div
    className="headerItem"
  >
    <p />
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
